### PR TITLE
feat(task): separate resource state from job

### PR DESCRIPTION
**Pull request by LioRoger**
#### What type of PR is this?

type-feature
#### What this PR does / why we need it:
this PR split the binding relationship of resource and job.
when job finished, it won't wait resource destroy.
resource became a transparent concept for job submitter

#### Which issue(s) this PR fixes:

Fixes #
1. refactor task module package  organization. make task top level of the package
2. make Odc Agent as independent class.
3. introduce resource mock mode for local development
4. task will not concern about resource anymore.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-server/src/main/java/com/oceanbase/odc/server/OdcServer.java [hunk 1]
```diff
@@ -81,14 +76,6 @@ public OdcServer(@Qualifier("metadbMigrate") AbstractMetaDBMigrate metadbMigrate
      * @param args
      */
     public static void main(String[] args) {
-        if (Objects.equals(SystemUtils.getEnvOrProperty(JobEnvKeyConstants.ODC_BOOT_MODE),
```
Comment (yhilmare) [inline server/odc-server/src/main/java/com/oceanbase/odc/server/OdcServer.java:84]: great job

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/BaseJobCaller.java [hunk 2]
```diff
@@ -185,8 +196,6 @@ protected void updateExecutorDestroyed(JobIdentity ji) throws JobException {
 
     protected abstract void doStop(JobIdentity ji) throws JobException;
 
-    protected abstract void doDestroyInternal(ExecutorIdentifier identifier) throws JobException;
-
-    protected abstract boolean isExecutorExist(ExecutorIdentifier identifier) throws JobException;
-
+    protected abstract boolean isExecutorExist(ExecutorIdentifier identifier, ResourceID resourceID)
```
[the remark this answers is not in the document]
Comment (LioRoger) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/BaseJobCaller.java:199]: There are several main consideration based on compatibility and extending.
1. ExecutorIdentifier contains host and port but resource ID not contains this information. 
2. Resource has state, if resource state change to destroying. the executor considered as destroyed. identifier can't represent the meaning.
3. k8s resource used resource id to found if resource state has changed and process resource used identifier to found if process id existed.

**Review by guowl3 — APPROVED** [review]
lgtm

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Review by LuckyPickleZZ — APPROVED** [review]

**Pull request merged**